src/rules/no-firestore-object-arrays.ts

```
@@ -15,46 +15,151 @@ const PRIMITIVE_TYPES = new Set([
+};
+
+const unwrapArrayElementType = (node: TSESTree.TypeNode): TSESTree.TypeNode => {
+  let current: TSESTree.TypeNode = node;
+  // Fixpoint loop: peel wrappers in any order until none remain
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    if (
+      current.type === AST_NODE_TYPES.TSTypeOperator &&
+      (current as TSESTree.TSTypeOperator).operator === 'readonly'
+    ) {
+      current = (current as TSESTree.TSTypeOperator).typeAnnotation as TSESTree.TypeNode;
```

ESLint

[error] Insert `⏎········`

(prettier/prettier)